aspnet/mvc/controllers/routing.rst

```diff
@@ -1,8 +1,738 @@
+   var routes = new RouteBuilder(app);
+
+   // Add connection to MVC, will be hooked up by calls to MapRoute.
+   routes.DefaultHandler = new MvcRouteHandler(...);
+
+   // Execute callback to register routes.
+   // routes.MapRoute("default", "{controller=Home}/{action=Index}/{id?}");
+
+   // Create route collection and add the middleware.
+   app.UseRouter(routes.Build());
+
+**[review: this section rewritten]**
+:dn:method:`~Microsoft.AspNetCore.Builder.MvcApplicationBuilderExtensions.UseMvc` does not directly define any routes, it adds a placeholder to the route collection for the ``attribute`` route. The overload ``UseMvc(Action<IRouteBuilder>)`` lets you add your own routes and also supports attribute routing. :dn:method:`~Microsoft.AspNetCore.Builder.MvcApplicationBuilderExtensions.UseMvcWithDefaultRoute` defines a default route and supports attribute routing. The :ref:`attribute-routing-ref-label` section includes more details on attribute routing.
+
+.. _routing-conventional-ref-label:
+
+Conventional routing
+---------------------
+
+The ``default`` route::
+
+  routes.MapRoute("default", "{controller=Home}/{action=Index}/{id?}");
+
+is an example of a *conventional routing*. We call this style *conventional routing* because it establishes a *convention* for URL paths:
+
+-  the first path segment maps to the controller name
+-  the second maps to the action name.
+-  the third segment is used for an optional ``id`` used to map to a model entity
+
+Using this ``default`` route, the URL path ``/Products/List`` maps to the ``ProductsController.List`` action, and ``/Blog/Article/17`` maps to ``BlogController.Article``. This mapping is based on the controller and action names **only** and is not based on namespaces, source file locations, or method parameters.
+
+.. Tip:: Using conventional routing with the default route allows you to build the application quickly without having to come up with a new URL pattern for each action you define. For an application with CRUD style actions, having consistency for the URLs across your controllers can help simplify .. review [things -  Edit replaced with following] your code and make your UI more predictable.
+
+.. warning: The ``id`` is defined as optional by the route template, meaning that your actions can execute *without* the ``id`` provided as part of the URL. [Original - see replacement below: Usually what will happen if ``id`` is omitted from the URL is that it will be set to ``0`` by model binding, and as a result no entity will be found in the database matching ``id == 0``.][{are there exceptions, for example can ambient values be used - if so we need to list exceptions and tell them they need to use attribute routing to prevent getting the wrong ID} ] Attribute routing can give you fine-grained control to make the id required for some actions and not for others. By convention the documentation will include optional parameters like ``id`` when they are likely to appear in correct usage.
```

Comment:
There are no exceptions to this. Ambient values are only a concept when considering **generating** URLs